core/src/main/java/com/scalar/db/io/DateColumn.java

```diff
@@ -0,0 +1,138 @@
+package com.scalar.db.io;
+
+import com.google.common.base.MoreObjects;
+import com.google.common.collect.ComparisonChain;
+import com.scalar.db.common.error.CoreError;
+import java.time.LocalDate;
+import java.util.Comparator;
+import java.util.Objects;
+import java.util.Optional;
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.Immutable;
+
+/**
+ * A {@code Column} for a DATE type. It represents a date without a time-zone in the ISO-8601
+ * calendar system, such as 2007-12-03
+ */
+@Immutable
+public class DateColumn implements Column<LocalDate> {
+  /** The minimum DATE value is 1000-01-01 */
+  public static final LocalDate MIN_VALUE = LocalDate.of(1000, 1, 1);
+  /** The maximum DATE value is 9999-12-31 */
+  public static final LocalDate MAX_VALUE = LocalDate.of(9999, 12, 31);
+
+  private final String name;
+  @Nullable private final LocalDate value;
+
+  private DateColumn(String name, @Nullable LocalDate value) {
+    if (value != null && (value.isBefore(MIN_VALUE) || value.isAfter(MAX_VALUE))) {
+      throw new IllegalArgumentException(
+          CoreError.OUT_OF_RANGE_COLUMN_VALUE_FOR_DATE.buildMessage(value));
+    }
+
+    this.name = Objects.requireNonNull(name);
+    this.value = value;
+  }
+
+  @Override
+  protected final void finalize() throws Throwable {
+    // TODO delete this method once https://github.com/scalar-labs/scalardb/pull/2421 is merge
```

Comment:
Thanks, I will do that in a separate PR because rebasing the feature branch `[add_time_related_types](https://github.com/scalar-labs/scalardb/tree/add_time_related_types)` now would mess up this PR I think 🤔